### PR TITLE
(1password) Update to fetch latest Versions & Beta

### PR DESCRIPTION
**Pull request by RedBaron2**
@gep13 @AdmiringWorm @majkinetor @pascalberger 
Adjustment of the fetching of the versions for the stream; Added the Before Update block to make for local use without errors/multi dependencies in the nuspec file; added the beta versions to the package.

## Description

This will fetch the OPW4, OPW7, & OPW7 beta versions. Since the latest link for OPW6 also is following the OPW7 listing. It is not listed but can be added should it diverge in the future.

## Motivation and Context
Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
This fixes the issue #1048 

## How Has this Been Tested?
Please describe in detail how you tested your changes.
Tested locally and via Appveyor

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:

- [X] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [X] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md)
- [ ] The added/modified package passed install/uninstall in the chocolatey test environment.
- [X] The changes only affect a single package (not including meta package).

## Original Location
- [Original Repository](add_link_to_original_repository_location)
- [Open Issues](link_to_the_generic_location_of_open_issues) *Add the different issues underneath, and tick those that are fixed in this PR*
  - [ ] Issue 1 link
  - [ ] Issue 2 Link
- [ ] *Include the link to the opened PR that removes the package from the original location*
- [ ] The [migration guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Package-migration-process) have been followed

### CODE REVIEW

File: automatic/1password/update.ps1 [hunk 1]
```diff
@@ -15,36 +14,50 @@ function global:au_SearchReplace {
...
-  }
+  } 
 }
 
 function global:au_GetLatest {
-  $download_page = Invoke-WebRequest -Uri $releases -UseBasicParsing
 
   $streams = @{}
-  $res = @('OPW4';'format=zip$')
+  $res = @('OPWY';'OPW4';'OPW7')
```
Comment (AdmiringWorm) [inline automatic/1password/update.ps1]: What is the `OPWY`?

File: automatic/1password/update.ps1 [hunk 2]
```diff
@@ -15,36 +14,50 @@ function global:au_SearchReplace {
   }
 }
 
-function global:au_AfterUpdate {
+function global:au_BeforeUpdate {
   . "$PSScriptRoot/update_helper.ps1"
-  if ($Latest.PackageName -eq '1password4') {
     removeDependencies ".\*.nuspec"
```
Comment (AdmiringWorm) [inline automatic/1password/update.ps1]: You can't do this in the `au_BeforeUpdate`, the changes will be overwritten when `au_SearchReplace` have ran.
It must be done in the `au_AfterUpdate` for the changes to be kept, please revert these changes.

File: automatic/1password/update.ps1 [hunk 3]
```diff
@@ -15,36 +14,50 @@ function global:au_SearchReplace {
...
+        'OPWY' {
+            $url32 = 'https://app-updates.agilebits.com/download/OPW7/Y'
+			$kind = '1password'
+        }
+    }
     $url32 = Get-RedirectedUrl $url32
-
     $verRe = '[-]|\.exe$'
     $version = $url32 -split $verRe | select -last 1 -skip 1
+    $version = $version -replace('\.BETA',' beta')
```
Comment (AdmiringWorm) [inline automatic/1password/update.ps1]: I don't see any how the urls actually being beta urls.
What's the format of the version from before this replacement is used?
`7.3.5.BETA`? If so, it's okay, but if not I think we'll remove this parsing for now, and wait until it's needed.

File: automatic/1password/update.ps1 (hunk 1 above)
Comment (RedBaron2) [reply automatic/1password/update.ps1]: It is a shorten version of the code they use for beta `OPW7/Y`

File: automatic/1password/update.ps1 (hunk 2 above)
Comment (RedBaron2) [reply automatic/1password/update.ps1]: This was done to make it so `'1password' already has a dependency defined for 'dotnet4.6.1
Choco pack failed with exit code 1` doesn't happen locally outside of `Github`

This is also what happens in the nuspec file `<dependencies><dependency id="dotnet4.6.1" version="4.6.01055.20170308" /><dependency id="dotnet4.6.1" version="4.6.01055.20170308" /></dependencies>`

This fix prevents this error from occurring

File: automatic/1password/update.ps1 (hunk 3 above)
Comment (RedBaron2) [reply automatic/1password/update.ps1]: Yes, the download URL(s)
1. `https://c.1password.com/dist/1P/win6/1PasswordSetup-7.1.566.BETA.exe`
2. `https://c.1password.com/dist/1P/win6/1PasswordSetup-7.0.558.BETA.exe`
3. `https://c.1password.com/dist/1P/win6/1PasswordSetup-7.0.554.BETA.exe`
4. `https://c.1password.com/dist/1P/win6/1PasswordSetup-7.0.543.BETA.exe`
for the previous beta(s)

**Comment by RedBaron2** [comment]
Will work on another way to get the links necessary to update the package, soon

File: automatic/1password/update.ps1 (hunk 2 above)
Comment (AdmiringWorm) [reply automatic/1password/update.ps1]: fair enough, but this style won't work either.
Here is what is going to be happening.
1. au_BeforeUpdate will remove the dependency, then save the nuspec file.
2. au_SearchReplace will run and save the file with the xml data that AU have cached, meaning it won't have detected the change that occurred during au_BeforeUpdate.

Comment (RedBaron2) [reply automatic/1password/update.ps1]: removed due to inconsistent results

**Comment by RedBaron2** [comment]
@AdmiringWorm 
Requested changes have been made. The file has also been updated

**Comment by RedBaron2** [comment]
@AdmiringWorm 

Thanks 😄
